### PR TITLE
Workaround for 500 Internal Error returned when Jenkins fails a healt…

### PR DESCRIPTION
**Pull request by harnitsignalfx**
…h check

### CODE REVIEW

**Comment by harnitsignalfx** [comment]
This is what happens when you hit the health check endpoint and a component is unhealthy: 

Notice that the json response is returned alongside of the 500 response

curl -v http://127.0.0.1:8080/metrics/<Metrics-Key>/healthcheck
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to 127.0.0.1 (127.0.0.1) port 8080 (#0)
> GET /metrics/<metrics-key>/healthcheck HTTP/1.1
> Host: 127.0.0.1:8080
> User-Agent: curl/7.53.1
> Accept: */*
> 
< HTTP/1.1 500 Server Error
< Date: Sun, 28 Jan 2018 05:01:14 GMT
< X-Content-Type-Options: nosniff
< Content-Type: application/json
< Cache-Control: must-revalidate,private,max-age=56
< Expires: Sun, 28 Jan 2018 05:02:10 GMT
< Last-Modified: Sun, 28 Jan 2018 05:01:10 GMT
< Content-Length: 183
< Server: Jetty(9.4.z-SNAPSHOT)
< 
* Connection #0 to host 127.0.0.1 left intact
{"disk-space":{"healthy":true},"plugins":{"healthy":false,"message":"There are 1 failed plugins: dockerhub.hpi"},"temporary-space":{"healthy":true},"thread-deadlock":{"healthy":true}}

File: jenkins.py [hunk 1]
```diff
@@ -106,7 +106,16 @@ def _api_call(url, opener, http_timeout):
     try:
         urllib2.install_opener(opener)
         resp = urllib2.urlopen(url, timeout=http_timeout)
-    except (urllib2.HTTPError, urllib2.URLError) as e:
+    except (urllib2.HTTPError) as e:
+	# Workaround for Jenkins issue with health check
+	# which returns 500 server code if a health check fails
+	# https://github.com/sensu-plugins/sensu-plugins-jenkins/issues/10
+	if e.code == 500:
+	    return json.loads(e.read())
```
Comment (keitwb) [inline jenkins.py]: I would not just blindly assume that all 500 responses have valid json.  Wrap the `json.loads` call with another try/except block and just log the standard error message if json parsing fails.

File: jenkins.py [hunk 2]
```diff
@@ -106,7 +106,16 @@ def _api_call(url, opener, http_timeout):
...
+	else:
```
Comment (keitwb) [inline jenkins.py]: Remove else statement and outdent the below log and return statement.

**Comment by nishantagarwal1990** [comment]
Fixing in #11

**Pull request closed without merging**